File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    const foods = []
    for(let i=1; i<food.length; i++){
        let cnt = parseInt(food[i] / 2) // 한 사람이 먹을 양
        if(cnt) foods.push(String(i).repeat(cnt))
    }
    let answer = foods.join('') + '0' + foods.reverse().join('')
    return answer;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    foods = [] # 한사람당 먹을 수 있는 양
    for i in range(1, len(food)):
        cnt = food[i] // 2
        if(cnt!=0): foods.append(str(i) * cnt)
    
    answer = ''.join(foods) + '0' + ''.join(reversed(foods))
    return answer